uint32 milliseconds
int32 rate
float32[] data